src/main/java/com/puppycrawl/tools/checkstyle/JavadocPropertiesGenerator.java

```diff
@@ -0,0 +1,321 @@
+     * @throws CheckstyleException if an unsupported inline tag found
+     */
+    private static String getFirstJavadocSentence(DetailNode tree) throws CheckstyleException {
+        String firstSentence = null;
+        final StringBuilder builder = new StringBuilder(128);
+        for (DetailNode node : tree.getChildren()) {
+            if (node.getType() == JavadocTokenTypes.TEXT) {
+                final Matcher matcher = END_OF_SENTENCE_PATTERN.matcher(node.getText());
+                if (matcher.find()) {
+                    // Commit the sentence if the end-of-sentence marker is found.
+                    firstSentence = builder.append(matcher.group(1)).toString();
+                    break;
+                }
+                // Otherwise append the whole line and look for the end-of-sentence marker
+                // on the next line.
+                builder.append(node.getText());
+            }
+            else if (node.getType() == JavadocTokenTypes.JAVADOC_INLINE_TAG) {
+                formatInlineCodeTag(builder, node);
+            }
+            else if (node.getType() == JavadocTokenTypes.HTML_ELEMENT) {
+                formatHtmlElement(builder, node);
+            }
+        }
+        return firstSentence;
+    }
+
+    /**
+     * Converts inline code tag into HTML form.
+     * @param builder to append
+     * @param inlineTag to format
+     * @throws CheckstyleException if the inline javadoc tag is not a literal nor a code tag
+     */
+    private static void formatInlineCodeTag(StringBuilder builder, DetailNode inlineTag)
+            throws CheckstyleException {
+        boolean isCodeTag = false;
+        for (DetailNode node : inlineTag.getChildren()) {
+            switch (node.getType()) {
+                // Ignore opening and closing curly braces and spaces after the tag.
+                case JavadocTokenTypes.JAVADOC_INLINE_TAG_START:
+                case JavadocTokenTypes.JAVADOC_INLINE_TAG_END:
+                case JavadocTokenTypes.WS:
+                    break;
+                // The text to append.
+                case JavadocTokenTypes.TEXT:
+                    if (isCodeTag) {
+                        builder.append("<code>").append(node.getText()).append("</code>");
+                    }
+                    else {
+                        builder.append(node.getText());
+                    }
+                    break;
+
+                // Supported tags.
+                case JavadocTokenTypes.LITERAL_LITERAL:
+                    break;
+                case JavadocTokenTypes.CODE_LITERAL:
+                    isCodeTag = true;
+                    break;
+
+                // Unsupported tags.
+                default:
+                    throw new CheckstyleException("Inline tag "
+                        + JavadocUtils.getTokenName(node.getType()) + " is not supported");
+            }
+        }
+    }
+
+    /**
+     * Rebuilds HTML text from the AST of a HTML_ELEMENT.
```

Comment:
"concatenate HTML text from AST of a JavadocTokenTypes.HTML_ELEMENT"

Reply:
done